www/class/centreon-clapi/CentreonLDAPContactRelation.class.php

```diff
@@ -0,0 +1,183 @@
+ * combined work based on this program. Thus, the terms and conditions of the GNU
+ * General Public License cover the whole combination.
+ *
+ * As a special exception, the copyright holders of this program give CENTREON
+ * permission to link this program with independent modules to produce an executable,
+ * regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of CENTREON choice, provided that
+ * CENTREON also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+namespace CentreonClapi;
+
+require_once "centreonObject.class.php";
+require_once "centreonUtils.class.php";
+require_once "centreonTimePeriod.class.php";
+require_once "Centreon/Object/Contact/Contact.php";
+require_once "Centreon/Object/Command/Command.php";
+require_once "Centreon/Object/Timezone/Timezone.php";
+require_once "Centreon/Object/Relation/Contact/Command/Host.php";
+require_once "Centreon/Object/Relation/Contact/Command/Service.php";
+require_once "centreonLDAP.class.php";
+
+/**
+ * Class representating relation between a contact and his LDAP configuration
+ *
+ * @author sylvestre
+ */
+class CentreonLDAPContactRelation extends CentreonObject
+{
+    const ORDER_UNIQUENAME = 1;
+    const ORDER_NAME = 0;
+    const ORDER_MAIL = 2;
+    const ORDER_PASS = 3;
+    const ORDER_ADMIN = 4;
+    const ORDER_ACCESS = 5;
+    const ORDER_LANG = 6;
+    const ORDER_AUTHTYPE = 7;
+    const HOST_NOTIF_TP = "hostnotifperiod";
+    const SVC_NOTIF_TP = "svcnotifperiod";
+    const HOST_NOTIF_CMD = "hostnotifcmd";
+    const SVC_NOTIF_CMD = "svcnotifcmd";
+    const UNKNOWN_LOCALE = "Invalid locale";
+    const UNKNOWN_TIMEZONE = "Invalid timezone";
+    const CONTACT_LOCATION = "timezone";
+    const UNKNOWN_NOTIFICATION_OPTIONS = "Invalid notifications options";
+    private const LDAP_PARAMETER_NAME = "ar_name";
+
+    protected $register;
+    public static $aDepends = array(
+        'CONTACT',
+        'LDAP'
+    );
+
+    /**
+     * Constructor
+     *
+     * @return void
+     */
+    public function __construct(\Pimple\Container $dependencyInjector)
+    {
+        parent::__construct($dependencyInjector);
+        $this->dependencyInjector = $dependencyInjector;
+        $this->tpObject = new CentreonTimePeriod($dependencyInjector);
+        $this->ldap = new CentreonLdap($dependencyInjector);
+        $this->object = new \Centreon_Object_Contact($dependencyInjector);
+        $this->timezoneObject = new \Centreon_Object_Timezone($dependencyInjector);
+        $this->params = array(
```

Comment:
please remove all useless code in this class
do not copy paste everything from contact class

Reply:
saved the constants ORDER_NAME, LDAP_PARAMETER_NAME and the constructor